Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/ActorAsset.cpp

```diff
@@ -74,49 +73,30 @@ namespace AZ
-                    uint32_t influenceIndex = 0;
 
-                    AZStd::vector<uint32_t> localIndices;
-                    AZStd::vector<float> localWeights;
+                    uint32_t influenceIndex = 0;
                     for (; influenceIndex < influenceCount; ++influenceIndex)
                     {
                         EMotionFX::SkinInfluence* influence = sourceSkinningInfo->GetInfluence(originalVertex, influenceIndex);
-                        localIndices.push_back(static_cast<uint32_t>(influence->GetNodeNr()));
+                        if (influenceIndex % 2 == 0)
+                        {
+                            blendIndexBufferData.push_back(static_cast<uint32_t>(influence->GetNodeNr()) << 16);
+                        }
+                        else
+                        {
+                            blendIndexBufferData.back() |= static_cast<uint32_t>(influence->GetNodeNr());
+                        }
                         blendWeightBufferData.push_back(influence->GetWeight());
                     }
 
                     // Zero out any unused ids/weights
                     for (; influenceIndex < maxInfluencesPerVertex; ++influenceIndex)
                     {
-                        localIndices.push_back(0);
-                        blendWeightBufferData.push_back(0.0f);
-                    }
-
-
-                    // [TODO ATOM-15288]
-                    // Temporary workaround. If there is cloth data, set all the blend weights to zero to indicate
-                    // the vertices will be updated by cpu. When meshes with cloth data are not dispatched for skinning
-                    // this can be hasClothData can be removed.
-
-                    // If there is no skinning info, default to 0 weights and display an error
-                    if (hasClothData || !sourceSkinningInfo)
-                    {
-                        for (influenceIndex = 0; influenceIndex < maxInfluencesPerVertex; ++influenceIndex)
+                        if (influenceIndex % 2 == 0)
                         {
-                            blendWeightBufferData[influenceIndex] = 0.0f;
+                            blendIndexBufferData.push_back(0);
                         }
-                    }
-
-                    // Make sure we have an even number of indices for packing into 32-bit uints
-                    if (localIndices.size() % 2 != 0)
-                    {
-                        localIndices.push_back(0);
-                    }
-
-                    // Now that we have the 16-bit indices, pack them into 32-bit uints
-                    for (size_t i = 0; i < localIndices.size(); i+=2)
-                    {
-                        blendIndexBufferData.push_back(localIndices[i] << 16 | localIndices[i+1]);
+                        blendWeightBufferData.push_back(0.0f);
```

Comment:
You can do this with `reisze()` directly:
```suggestion
                        blendWeightBufferData.resize(maxInfluencesPerVertex, 0.0f);
```

Reply:
I can't exactly - this isn't a local vector with just the weights for this vertex, it has all the prior weights as well. I could do something like blendWeightBufferData.resize(blendWeightBufferData.size() + maxInfluencesPerVertex - influenceIndex, 0.0f), but this seems simpler to me.